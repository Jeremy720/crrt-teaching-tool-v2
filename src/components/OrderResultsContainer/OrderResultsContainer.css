.OrderResultsContainer {
  background-color: #fff;
  border: 2px solid #efefef;
  border-radius: 10px;
  height: auto;
  margin: 2rem auto;
  max-height: 90%;
  overflow: scroll;
  width: 50%;
  display: grid;
  grid-template-columns: 2rem 1fr;
}

.orders-results-sidebar {
  background-color: #4ea84e;
  border-radius: 5px 0 0 5px;
}

.orders-results-main-content {
  padding: 0 1rem;
}
