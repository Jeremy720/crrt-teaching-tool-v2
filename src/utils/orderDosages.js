const orderDosages = {
  requiredRanges: {
    "sodium": {
      min: 130,
      max: 150
    },
    "potassium": {
      max: 4
    },
    "chloride": {
      min: 95,
      max: 110
    },
    "bicarbonate": {
      min: 20,
      max: 40
    },
    "calcium": {
      min: 0,
      max: 3
    },
    "magnesium": {
      min: 0,
      max: 2
    },
    "phosphorous": {
      min: 0,
      max: 1
    },
    "grossUltraFiltration": {
      min: 0,
      max: 2000
    },
    "replacementFluidFlowRate": {
      min: 0,
      max: 8
    },
  },
  errorMessages: {
    "sodium": "The hospital pharmacy can only compound sodium between 130 and 150.  Use D5W or 3% normal saline if needed",
    "potassium": "The hospital pharmacy will not compound fluid with a potassium above 4 mg/dL",
    "chloride": "The hospital pharmacy can only compound chloride between 95 and 110",
    "bicarbonate": "The hospital pharmacy can only compound bicarbonate between 20 and 40",
    "calcium": "The hospital pharmacy can only compound calcium between 0 and 3 mEq/L",
    "magnesium": "The hospital pharmacy can only compound magnesium between 0 and 2 mgd/L",
    "phosphorous": "The hospital pharmacy cannot compound fluid with > 1mg/dL of phosphorous, due to precipitation of calcium phosphate.",
    "grossUltraFiltration": "The CRRT machine will not accept ultrafiltration rates above 2,000 mL/hour",
    "replacementFluidFlowRate": "The CRRT machine cannot deliver fluid above 8 L/hr",
  }
}

export default orderDosages;