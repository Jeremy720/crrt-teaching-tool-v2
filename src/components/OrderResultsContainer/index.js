import React from 'react';
import OrderResultsCard from '../OrderResultsCard';
import './OrderResultsContainer.css';
const uuidv4 = require('uuid/v4');

const OrderResultsContainer = ({ orderResults }) => {
	const mockData = [
		{
			timeStamp: '6:00 PM - Day 1',
			mockMessages: [
			'The patient’s filter clotted twice, and was replaced.',
			'The patient’s filter clotted twice, and was replaced.',
			'The patient’s filter clotted twice, and was replaced.',
			'The patient’s filter clotted twice, and was replaced.',
			'The patient’s filter clotted twice, and was replaced.',
			'The patient’s filter clotted twice, and was replaced.'
			]
		},
		{
			timeStamp: '10:00 AM - Day 1',
			mockMessages: [
			'Welcome to the CRRT education application simulator!',
			'Welcome to the CRRT education application simulator!',
			'Welcome to the CRRT education application simulator!'
			]
		}
	]

	// const orderResultsCards = orderResults.map(result => {
	// 	return <OrderResultsCard key={uuidv4()} {...result}/>
	// })

	const mockResultsCards = mockData.map(object => {
		const { timeStamp, mockMessages } = object
		return <OrderResultsCard key={uuidv4()} timeStamp={timeStamp} messages={mockMessages}/>
	})


	return (
		<div className='OrderResultsContainer'>
			<div className='orders-results-sidebar'>
			</div>
			<div className='orders-results-main-content'>
				{/* { orderResultsCards } */}
				{ mockResultsCards }
			</div>
		</div>
	)
}

export default OrderResultsContainer;