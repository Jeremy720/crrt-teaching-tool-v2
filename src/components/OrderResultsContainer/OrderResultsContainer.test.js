import React from 'react';
import OrderResultsContainer from './';
import { shallow } from 'enzyme';
import orderDosages from '../../utils/orderDosages.js';

describe('OrderResultsContainer', () => {
  let wrapper;
  let mockState;

  beforeEach(() => {
    wrapper = shallow(<OrderResultsContainer />);
  });

  it('Should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});