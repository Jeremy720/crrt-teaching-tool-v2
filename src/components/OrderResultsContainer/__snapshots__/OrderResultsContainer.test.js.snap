// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderResultsContainer Should render like snapshot 1`] = `undefined`;
